text/0000-cargo-embed-dependency-versions.md

```diff
@@ -0,0 +1,184 @@
+            "required": [
+                "name",
+                "source",
+                "version"
+            ],
+            "properties": {
+                "dependencies": {
+                    "description": "Packages are stored in an ordered array both in the `VersionInfo` struct and in JSON. Here we refer to each package by its index in the array. May be omitted if the list is empty.",
+                    "type": "array",
+                    "items": {
+                        "type": "integer",
+                        "format": "uint",
+                        "minimum": 0.0
+                    }
+                },
+                "kind": {
+                    "description": "\"build\" or \"runtime\". May be omitted if set to \"runtime\". If it's both a build and a runtime dependency, \"runtime\" is recorded.",
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/DependencyKind"
+                        }
+                    ]
+                },
+                "name": {
+                    "description": "Crate name specified in the `name` field in Cargo.toml file. Examples: \"libc\", \"rand\"",
+                    "type": "string"
+                },
+                "root": {
+                    "description": "Whether this is the root package in the dependency tree. There should only be one root package. May be omitted if set to `false`.",
+                    "type": "boolean"
+                },
+                "source": {
+                    "description": "Currently \"git\", \"local\", \"crates.io\" or \"registry\". May be extended in the future with other revision control systems, etc.",
+                    "allOf": [
+                        {
+                            "$ref": "#/definitions/Source"
+                        }
+                    ]
+                },
+                "version": {
+                    "description": "The package's version in the [semantic version](https://semver.org) format.",
+                    "type": "string"
+                }
+            }
+        },
+        "Source": {
+            "description": "Serializes to \"git\", \"local\", \"crates.io\" or \"registry\". May be extended in the future with other revision control systems, etc.",
+            "oneOf": [
+                {
+                    "type": "string",
+                    "enum": [
+                        "crates.io",
+                        "git",
+                        "local",
+                        "registry"
+                    ]
+                },
+            ]
+        }
+    }
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Slightly increases the size of the generated binaries. However, the increase is [typically below 1%](https://github.com/rust-lang/rfcs/pull/2801#issuecomment-549184251).
+- Adds more platform-specific code to the build process, which needs to be maintained.
+
+# Rationale and alternatives
```

Comment:
What about alternative [SBOM formats](https://github.com/rust-secure-code/cargo-auditable/issues/31)?

Reply:
They are not designed for this use case, and we found them unsuitable for a variety of reasons - the size too large, they require a timestamp that disrupts reproducible builds, etc.

[Trivy](https://github.com/aquasecurity/trivy) can be used to convert from this internal format to other SBOM formats, right now. I believe external tooling that converts between formats is the way to go. 

The proposed format is similar to other SBOMs in structure, so supporting it in tooling that deals with SBOMs is rather trivial (or at least it was in Trivy).

Reply:
Added to the RFC